cmake/SDL.cmake

```diff
@@ -71,7 +71,8 @@ if(UNIX)
     elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
         get_filename_component(CXX_CMD_NAME ${CMAKE_CXX_COMPILER} NAME)
         # Fujitsu CXX compiler does not support "-fstack-protector-all".
-        if(NOT CXX_CMD_NAME STREQUAL "FCC")
+        # The same applies to GPU targets (CUDA/HIP).
+        if(NOT (CXX_CMD_NAME STREQUAL "FCC" OR DNNL_SYCL_CUDA OR DNN_SYCL_HIP))
```

Comment:
We don't see any issues with CUDA backend when using `-fstack-protector-all`. Does it mean it's supported for CUDA and not supported for HIP?

Also, is there any way to keep the option for the host compiler while disabling it for the device compiler?

Reply:
>Does it mean it's supported for CUDA and not supported for HIP?

As far as I know, it's only because compiler will decide to ignore the option and warn during compilation, i.e: https://godbolt.org/z/nKeq5PaWP, but that's only my guess so take it with a pinch of salt.

> Also, is there any way to keep the option for the host compiler while disabling it for the device compiler?

As for your second question perhaps `sycl-target-backend` option could be of help here, see: https://intel.github.io/llvm-docs/UsersManual.html#target-toolchain-options
It would be something along the lines of `-fsycl-target-backend=amdgcn-amd-amdhsa -fno-stack-protector`. Again take it with a pinch of salt, I've not tested it myself.

In general feel free to reject/close this PR, I was trying to unblock compilation for HIP backend, as we've seen similar issues as discussed in https://github.com/intel/llvm/issues/7269 on our side, but as I don't know this codebase well enough, I'm happy to describe the problem (and potential solution) in an issue against https://github.com/oneapi-src/oneDNN.